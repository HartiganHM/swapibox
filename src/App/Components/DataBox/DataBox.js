import React from 'react';
import './DataBox.css';

const DataBox = () => {
    return(
        <div>
            I am duh DataBox!
        </div>
    )
}

export default DataBox;