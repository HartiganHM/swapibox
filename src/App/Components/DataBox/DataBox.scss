@import '../../../variables.scss';

.DataBox {
    align-items: stretch;
    background-color: $background-light;
    display: flex;
    flex: 3;
    flex-direction: column;
    height: inherit;
    padding: $padding-standard;
    position: relative;

    .header {
        align-items: center;
        border-bottom: $border-bottom;
        display: flex;
        font-size: 4.8rem;
        justify-content: space-around;
        padding: $padding-standard;
    }

    .button-box {
        display: flex;
        justify-content: space-around;
        padding:$padding-standard;
    }
}