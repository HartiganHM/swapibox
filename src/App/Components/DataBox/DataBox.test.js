import React from 'react';
import DataBox from './DataBox';
import { shallow } from 'enzyme';

describe('DataBox Tests', () => {
  let renderedDataBox;
  let mockDisplayData;
  let mockSelectData;
  let mockCurrentDisplay;
  let mockCurrentFavorites;
  let mockSaveFavorite;
  let mockRemoveFavorite;

  beforeEach(() => {
    mockDisplayData = [];
    mockSelectData = jest.fn();
    mockCurrentDisplay = null;
    mockCurrentFavorites = [];
    mockSaveFavorite = jest.fn();
    mockRemoveFavorite = jest.fn();

    renderedDataBox = shallow(
      <DataBox
        displayData={mockDisplayData}
        selectData={mockSelectData}
        currentDisplay={mockCurrentDisplay}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorite}
        removeFavorite={mockRemoveFavorite}
      />
    );
  });

  it('Should exist', () => {
    expect(renderedDataBox).toBeDefined();
  });

  it('Should match the snapshot', () => {
    expect(renderedDataBox).toMatchSnapshot();
  });

  it('Should have four Button components', () => {
    expect(renderedDataBox.find('Button').length).toEqual(4);
  });
});
