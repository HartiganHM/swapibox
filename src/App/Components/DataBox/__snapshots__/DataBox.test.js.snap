// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataBox Tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataBox
    currentDisplay={null}
    currentFavorites={Array []}
    displayData={Array []}
    removeFavorite={[Function]}
    saveFavorite={[Function]}
    selectData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          favoriteCount={undefined}
          selectData={[Function]}
          type="favorites button"
          value="View Favorites"
/>,
        <header
          className="header"
/>,
        <div
          className="button-box"
>
          <Button
                    currentDisplay={null}
                    selectData={[Function]}
                    type="category button"
                    value="People"
          />
          <Button
                    currentDisplay={null}
                    selectData={[Function]}
                    type="category button"
                    value="Planets"
          />
          <Button
                    currentDisplay={null}
                    selectData={[Function]}
                    type="category button"
                    value="Vehicles"
          />
</div>,
        <CardContainer
          category={Array []}
          currentDisplay={null}
          currentFavorites={Array []}
          removeFavorite={[Function]}
          saveFavorite={[Function]}
/>,
      ],
      "className": "DataBox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoriteCount": undefined,
          "selectData": [Function],
          "type": "favorites button",
          "value": "View Favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header",
        },
        "ref": null,
        "rendered": null,
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              currentDisplay={null}
              selectData={[Function]}
              type="category button"
              value="People"
/>,
            <Button
              currentDisplay={null}
              selectData={[Function]}
              type="category button"
              value="Planets"
/>,
            <Button
              currentDisplay={null}
              selectData={[Function]}
              type="category button"
              value="Vehicles"
/>,
          ],
          "className": "button-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "People",
            "nodeType": "function",
            "props": Object {
              "currentDisplay": null,
              "selectData": [Function],
              "type": "category button",
              "value": "People",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Planets",
            "nodeType": "function",
            "props": Object {
              "currentDisplay": null,
              "selectData": [Function],
              "type": "category button",
              "value": "Planets",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Vehicles",
            "nodeType": "function",
            "props": Object {
              "currentDisplay": null,
              "selectData": [Function],
              "type": "category button",
              "value": "Vehicles",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "category": Array [],
          "currentDisplay": null,
          "currentFavorites": Array [],
          "removeFavorite": [Function],
          "saveFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            favoriteCount={undefined}
            selectData={[Function]}
            type="favorites button"
            value="View Favorites"
/>,
          <header
            className="header"
/>,
          <div
            className="button-box"
>
            <Button
                        currentDisplay={null}
                        selectData={[Function]}
                        type="category button"
                        value="People"
            />
            <Button
                        currentDisplay={null}
                        selectData={[Function]}
                        type="category button"
                        value="Planets"
            />
            <Button
                        currentDisplay={null}
                        selectData={[Function]}
                        type="category button"
                        value="Vehicles"
            />
</div>,
          <CardContainer
            category={Array []}
            currentDisplay={null}
            currentFavorites={Array []}
            removeFavorite={[Function]}
            saveFavorite={[Function]}
/>,
        ],
        "className": "DataBox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoriteCount": undefined,
            "selectData": [Function],
            "type": "favorites button",
            "value": "View Favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header",
          },
          "ref": null,
          "rendered": null,
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                currentDisplay={null}
                selectData={[Function]}
                type="category button"
                value="People"
/>,
              <Button
                currentDisplay={null}
                selectData={[Function]}
                type="category button"
                value="Planets"
/>,
              <Button
                currentDisplay={null}
                selectData={[Function]}
                type="category button"
                value="Vehicles"
/>,
            ],
            "className": "button-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "People",
              "nodeType": "function",
              "props": Object {
                "currentDisplay": null,
                "selectData": [Function],
                "type": "category button",
                "value": "People",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Planets",
              "nodeType": "function",
              "props": Object {
                "currentDisplay": null,
                "selectData": [Function],
                "type": "category button",
                "value": "Planets",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Vehicles",
              "nodeType": "function",
              "props": Object {
                "currentDisplay": null,
                "selectData": [Function],
                "type": "category button",
                "value": "Vehicles",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "category": Array [],
            "currentDisplay": null,
            "currentFavorites": Array [],
            "removeFavorite": [Function],
            "saveFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
